File: src/web/src/models/solutionMatrix.ts
export interface Environment {
  id: string
  displayName: string
  url: string
}

export interface Solution {
  solutionId: string
  uniqueName: string
  friendlyName: string
  version: string
  isManaged: boolean
}

export interface SolutionDifferences {
  missing: boolean
  version: boolean
  managedState: boolean
}

export interface SolutionComparison {
  uniqueName: string
  friendlyName: string
  environments: Record<string, Solution | null>
  differences: SolutionDifferences
}

File: src/api/Models/SolutionMatrix.cs
namespace PowerPlatformDriftAnalyzer.Api.Models;

public sealed record AnalyzerEnvironment(string Id, string DisplayName, string Url);

public sealed record Solution(
    Guid SolutionId,
    string UniqueName,
    string FriendlyName,
    string Version,
    bool IsManaged);

public sealed record EnvironmentSnapshot(
    AnalyzerEnvironment Environment,
    IReadOnlyList<Solution> Solutions);

public sealed record SolutionDifferences(bool Missing, bool Version, bool ManagedState);

public sealed record SolutionComparison(
    string UniqueName,
    string FriendlyName,
    IReadOnlyDictionary<string, Solution?> Environments,
    SolutionDifferences Differences);

public sealed record CompareSolutionsRequest(string?[]? EnvironmentIds);
